shm/Map.cpp

```diff
@@ -1,29 +1,28 @@
 #include "Map.hpp"
 #include <algorithm>
 #include <random>
-#include <iostream>
+#include <cstddef>
 
 Map::Map(){
-    islandsLocations_.reserve(amountIslands);
-    generateIslands();
+    islandsLocations_.reserve(amountOfIslands);
+    void generateIslands(int numIslandsToGenerate);
 }
 
-void Map::generateIslands() {
+void Map::generateIslands(int numIslandsToGenerate) {
     std::random_device rd;
     std::mt19937 gen(rd());
-    std::uniform_int_distribution<> distrib(widthIsland, heightIsland);
+    std::uniform_int_distribution<> distrib(minPositionXY, maxPositionXY);
     
-      for (size_t i = 0; i < amountIslands; i++) { 
-        size_t positionX; 
-        size_t positionY; 
-        do { 
-            size_t positionX = distrib(gen); 
-            size_t positionY = distrib(gen);
-        } while (std::any_of(islandsLocations_.begin(), islandsLocations_.end(), [positionX, positionY](auto& island) { 
-            return island.getPosition() == Coordinates(positionX, positionY); 
-        })); 
-        islandsLocations_.push_back(Island(positionX, positionY)); 
-    }
+    do { 
+        size_t positionX = distrib(gen); 
+        size_t positionY = distrib(gen); 
+        if (std::any_of(islandsLocations_.begin(), islandsLocations_.end(), 
+                [positionX, positionY](auto& island) { 
+                return island.getPosition() == Coordinates(positionX, positionY); 
+            })) { 
+            islandsLocations_.push_back(Island(positionX, positionY));
+        } 
+    } while (amountOfIslands <10); 
```

Comment:
This loop will only execute once. amountOfIslands is constant and won't change. It's always 10.  Also, declaring position X and Y in each loop iteration is unnecessary.

```suggestion
        size_t positionX {}; 
        size_t positionY {};
    do { 
        positionX = distrib(gen); 
        positionY = distrib(gen); 
        if (std::any_of(islandsLocations_.begin(), islandsLocations_.end(), 
                [positionX, positionY](auto& island) { 
                return island.getPosition() == Coordinates(positionX, positionY); 
            })) { 
            islandsLocations_.push_back(Island(positionX, positionY));
        } 
    } while (numIslandsToGenerate-- > 0); 
```